### PR TITLE
Update to PHPUnit 10

### PR DESCRIPTION
**Pull request by Slamdunk**

### CODE REVIEW

File: psalm-baseline.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<files psalm-version="5.7.7@e028ba46ba0d7f9a78bc3201c251e137383e145f">
+  <file src="test/StoragelessTest/Session/LazySessionTest.php">
+    <InvalidFunctionCall>
+      <code><![CDATA[($this->sessionLoader)()]]></code>
```
Comment (Slamdunk) [inline psalm-baseline.xml]: So, when I move from:
```php
    /** @var MakeSession&MockObject */
    private MakeSession $sessionLoader;
```
to
```php
    /** @var MakeSession&MockObject */
    private MakeSession&MockObject $sessionLoader;
```
Tests of course pass, and PSalm doesn't need the baseline.

But then `phpcbf` removed the DocBlock, and for some unbeknownst reason, PSalm breaks :cry:

Comment (Ocramius) [reply psalm-baseline.xml]: I'd prioritize psalm over phocbf here: we can tell phpcbf to shut up here, and remove the baseline instead.

Rationale: this project should remain baseline -less.

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Slamdunk** [comment]
@Ocramius I bet the infection fail is related to PHPUnit 10 completely dropped `phpdbg` support, and `laminas-continuous-integration-action` not installing PCOV nor xDebug, but I'm not aware on how exactly `laminas-ci` works.

Can you tell me where to look to figure this out?

**Comment by Ocramius** [comment]
That's on me to do: https://github.com/laminas/laminas-continuous-integration-action/issues/130

**Pull request merged**